File: feign-core/src/main/java/name/abhijitsarkar/feign/core/model/Method.java
/*
 * Copyright (c) 2016, the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * License for more details.
 */

package name.abhijitsarkar.feign.core.model;

import lombok.Getter;

import static java.lang.Boolean.FALSE;

/**
 * @author Abhijit Sarkar
 */
@Getter
@SuppressWarnings({"PMD.BeanMembersShouldSerialize", "PMD.SingularField"})
public class Method {
    public static final String WILDCARD = ".*";

    private String name;

    private Boolean ignoreCase;

    public Method() {
        setName(null);
        setIgnoreCase(null);
    }

    public Boolean isIgnoreCase() {
        return ignoreCase;
    }

    public void setName(String name) {
        this.name = (name == null) ? WILDCARD : name;
    }

    public void setIgnoreCase(Boolean ignoreCase) {
        this.ignoreCase = (ignoreCase == null) ? FALSE : ignoreCase;
    }
}

File: feign-core/src/main/scala/name/abhijitsarkar/feign/core/model/Method.scala
/*
 * Copyright (c) 2016, the original author or authors.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; version 3 of the License.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * A copy of the GNU General Public License accompanies this software,
 * and is also available at http://www.gnu.org/licenses.
 */

package name.abhijitsarkar.feign.core.model

import java.lang.{Boolean => JavaBoolean}

/**
  * @author Abhijit Sarkar
  */
class Method {
  var name: String = _
  var ignoreCase: JavaBoolean = _

  setName(null)
  setIgnoreCase(null)

  def getName = name

  import Method.WILDCARD

  def setName(name: String) {
    this.name = if (name == null) WILDCARD
    else name
  }

  def ignoreCaseToScala: Option[Boolean] = Option(ignoreCase)

  def getIgnoreCase = ignoreCase

  def setIgnoreCase(ignoreCase: JavaBoolean) {
    this.ignoreCase = if (ignoreCase == null) JavaBoolean.FALSE else ignoreCase
  }

  override def toString = s"Method($name, $ignoreCase)"
}

object Method {
  val WILDCARD = ".*"
}
